framework/web_app/src/pages/ProductSearch.vue

```diff
@@ -0,0 +1,197 @@
+                                {{ `${getOfferFullName(offer.brand, offer.name)} | ${getOfferSize(offer.size_value, offer.size_unit)}` }}
+                            </div>
+                            <div>
+                                <div class="text-body2 text-weight-medium">
+                                    {{ `$${offer.price_now?.toFixed(2).toString()}` }}
+                                </div>
+                                <div class="text-caption text-weight-regular">
+                                    {{
+                                        `was $${offer.price_was
+                                            ? offer.price_was?.toFixed(2).toString()
+                                            : offer.price_now?.toFixed(2).toString()}`
+                                    }}
+                                </div>
+                            </div>
+                        </div>
+                    </q-card-section>
+                </template>
+            </card-component>
+
+        </div>
+    </div>
+    <div
+        v-if="!Loading.isActive && previousSearchTerm && !productStore.productOffers?.length"
+        class="row text-h3 q-ma-sm items-center">
+        <img src="../../src/assets/banana-peel.jpg" class="q-pa-sm" style="border-radius: 50%; width: 230px;" />
+        <span class="text-h5 q-pa-sm">
+            No products found for '{{ previousSearchTerm }}'.
+        </span>
+    </div>
+</template>
+
+<script setup lang="ts">
+
+import { Loading } from 'quasar';
+import CardComponent from 'src/components/CardComponent.vue';
+import SelectComponent from 'src/components/SelectComponent.vue';
+import { IOfferSortByOption, OfferSortByOptions } from 'src/helpers/OfferSortByOptions';
+import { getOfferFullName, getOfferSize, isOfferFavourited } from 'src/helpers/ScrapedProductOfferLogic';
+import { Merchant } from 'src/models/Merchant';
+import ImageService from 'src/services/files/ImageService';
+import { useProductStore } from 'src/stores/ProductStore';
+import { ref } from 'vue';
+
+const productStore = useProductStore();
+
+//#region Search
+
+const searchTerm = ref('');
+const previousSearchTerm = ref<undefined | string>();
+
+let currentPage = 1;
+
+const search = (): Promise<string> =>
+    productStore.searchByTermAsync({
+        search_term: searchTerm.value,
+        start_page: currentPage
+    })
+    .then(() => previousSearchTerm.value = searchTerm.value);
+
+//#endregion Search
+
+//#region Filters
+
+const showFilters = ref(true);
+
+const toggleShowFilters = (): boolean =>
+    showFilters.value = !showFilters.value;
+
+const optionLabelSelector = (option: IOfferSortByOption): string =>
+    option.Description;
+
+//#endregion Filters
+
+//#region Offers
+
+const imageService = new ImageService();
+
+const colourByMerchant: { [key: string]: string } = {
+    Coles: 'red-14',
+    Woolworths: 'green-14'
+};
+
+const getMerchantColour = (merchantName: string): string =>
```

Comment:
This got superseded by something for logos instead.

There is UI or logic  i'll have to make reusable/shared once i do the my products page because it is similar to this page.

For now, i think it is fine.